symbulate/tests/test_distributions.py

```diff
@@ -10,25 +10,676 @@
+
+    def test_Geometric_to_NBinom(self):
+        exp_list, obs_list = [], []
+        X = Geometric(p=0.8)
+        sims = X.sim(Nsim)
+        simulated = sims.tabulate()
+        for k in range(1, 10):
+            expected = Nsim  * stats.nbinom(n=1,p=0.8).pmf(k-1)
+            if expected > 5:
+                exp_list.append(expected) 
+                obs_list.append(simulated[k]) 
+        pval = stats.chisquare(obs_list, exp_list).pvalue
+        self.assertTrue(pval > 0.01)
+
+
+class TestNegativeBinomial(unittest.TestCase):
+
+    def test_NBinom_error_r(self):
+        self.assertRaises(Exception, 
+                          lambda: NegativeBinomial(r=-10, p=0.6))
+    
+    def test_NBinom_p_1(self):
+        X = NegativeBinomial(r=10, p=1)
+        sims = X.sim(Nsim)
+        self.assertTrue(all(sim == 10 for sim in sims))
+    
+    def test_NBinom_Pascal_additive(self):
+        exp_list, obs_list = [], []
+        X, Y = RV(Pascal(r=4, p=0.6) * Pascal(r=6, p=0.6))
+        sims = (X + Y).sim(Nsim)
+        simulated = sims.tabulate()
+        for k in range(10, 35):
+            expected = Nsim * stats.nbinom(n=10, p=0.6).pmf(k)
+            if expected > 5:
+                exp_list.append(expected) 
+                obs_list.append(simulated[k])
+        pval = stats.chisquare(obs_list, exp_list).pvalue
+        self.assertTrue(pval > .01)
+
+    def test_NBinom_to_Geometric(self):
+        exp_list, obs_list = [], []
+        X = NegativeBinomial(r=1, p=0.8)
+        sims = X.sim(Nsim)
+        simulated = sims.tabulate()
+        for k in range(10):
+            expected = Nsim * stats.geom(p=0.8).pmf(k)
+            if expected > 5:
+                exp_list.append(expected)
+                obs_list.append(simulated[k])
+        pval  = stats.chisquare(obs_list, exp_list).pvalue
+        self.assertTrue(pval > 0.01)
+
+
+class TestPascal(unittest.TestCase):
+
+    def test_Pascal_error_r(self):
+        self.assertRaises(Exception, lambda: Pascal(r=0, p=0.3))
+    
+    def test_Pascal_p_1(self):
+        X = Pascal(r=10, p=1.0)
+        sims = X.sim(Nsim)
+        self.assertTrue(all(sim == 0 for sim in sims))
+
+
+class TestPoisson(unittest.TestCase):
+    
+    def test_Poisson_error(self):
+        self.assertRaises(Exception, lambda: Poisson(lam=0))
+
+    def test_Poisson_additive(self):
+        exp_list, obs_list = [], []
+        X, Y = RV(Poisson(lam=4) * Poisson(lam=7))
+        sims = (X + Y).sim(Nsim)
+        simulated = sims.tabulate()
+        for k in range(25):
+            expected = Nsim * stats.poisson(mu=11).pmf(k)
+            if expected > 5:
+                exp_list.append(expected)
+                obs_list.append(simulated[k])
+        pval = stats.chisquare(obs_list, exp_list).pvalue
         self.assertTrue(pval > .01)
 
+    def test_conditional_Poisson_add(self):
+        obs_list, exp_list = [], []
+        X,Y = RV(Poisson(lam=6) * Poisson(lam=7))
+        sims = (X|(X+Y == 8)).sim(Nsim)
```

Comment:
Why would you condition on X + Y == 8? This is an event with a less than 5% chance of happening for a Poisson(13). I would condition on X + Y == 11, which has a much higher probability.